packages/skill-lesson/trpc/routers/lessons.ts

```diff
@@ -31,22 +31,27 @@ export const lessonsRouter = router({
 
       const lessons = section ? section.lessons : module.lessons
 
-      const exerciseForSolution = lessons.find((resource: SanityDocument) => {
-        return resource.solution?._key === lesson.solution?._key
-      })
-
-      // TODO: this is likely overloading the lesson type just because it
-      //   exists and is different that the exercise type. We should probably
-      //   create a new type instead of `lesson` since `lesson` describes ALL
-      //   lessons, not just the ones that are not exercises.
-      const current =
-        input.type === 'lesson'
```

Comment:
we were looking for a special type here with `lesson` but that wasn't catching `explainer`